### ISSUE TITLE
LogOutputChannel not cleared properly

### ISSUE THREAD

**Issue by nhedger** [microsoft/vscode#204946]
Type: <b>Bug</b>

I'm building an extension that uses the LogOutputChannel to output some logs when the extension activates.

I started noticing that everytime I reloaded the window, I'd end up with the same logs twice: the logs from the previous activation, and the logs from the current activation. 

I assumed that the channel wasn't cleared automatically so at first I manually cleared the output channel using the context menu action before reloading the window. Unfortunately, this still results in the logs from the previous activation being shown. Programmatically clearing the output channel using the `clear` method did not work either.

The only way I could the channel to clear was to completely restart VS Code.

I'm thinking maybe the buffer is not being cleared and when the window is reloaded, the logs from the current activation are appended to the logs from the previous one.

VS Code version: Code 1.86.1 (31c37ee8f63491495ac49e43b8544550fbae4533, 2024-02-07T09:09:01.236Z)
OS version: Darwin x64 23.2.0
Modes:

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i9-9980HK CPU @ 2.40GHz (16 x 2400)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: disabled_off<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_graphite: disabled_off<br>video_decode: enabled<br>video_encode: enabled<br>webgl: enabled<br>webgl2: enabled<br>webgpu: enabled|
|Load (avg)|2, 3, 3|
|Memory (System)|32.00GB (7.05GB free)|
|Process Argv||
|Screen Reader|no|
|VM|0%|
</details>

**Comment by sandy081** [comment]
It is designed that logs are retained on window reload

**Comment by nhedger** [comment]
I understand persisting the logs between window reloads, but why persist them after having explicitly **cleared** them? This certainly feels counterintuitive.

**Comment by sandy081** [comment]
> but why persist them after having explicitly cleared them

Good point.

**Comment by nhedger** [comment]
Should this be reopened?

**Comment by Splines** [comment]
As discussed in #232500, the behavior with regards to the log file _written to disk_ is also different on WSL compared to MacOS:

- Inside an extension, create a [`LogOutputChannel`](https://code.visualstudio.com/api/references/vscode-api#LogOutputChannel) via:

```js
const logger = vscode.window.createOutputChannel("My extension", { log: true });
```

- Use `Developer: Open Log File` to open the log file for `My extension`.
- On Linux (WSL2 Ubuntu), we observe that after a `Developer: Reload Window`, when using `Open Log File` again, **a _new_ log file is opened** that VSCode neatly created for us in the folder `exthost<some_number+1>`. Here, `<some_number` is the number for the folder in which the log file resided prior to the window reload.
 - However, on MacOS (`Darwin 23.6.0 arm64`), we notice that **the _same_ logfile is reused** and lines are only appended to it. Is there a way we can force a new log file to be created on this system as well? Or at least that the old logfile is cleared? `logger.clear()` or `logger.dispose()` does not seem to work, it only clears the log from the output pane but does not affect the log file.

**Comment by nhedger** [comment]
Could you reopen this issue @ sandy081? You've acknowledged the problem, but it seems you forgot to reopen the issue. I'm afraid this won't ever get solved if it stays closed.

**Comment by nhedger** [comment]
This is not a feature request, but a bug report.

**Comment by nhedger** [comment]
Thanks for reopening @sandy081! I think the feature request tag should also be removed too, if you don't mind.

**Comment by sandy081** [comment]
I am sorry, this is not a bug, this is existing intended behaviour. To support this, we should support both behaviour. Hence tagged as feature request. 

Thanks

**Comment by nhedger** [comment]
I respectfully disagree. Explicitly clearing the logs does not actually clear them, which feels like a bug, wouldn't you agree?

**Comment by sandy081** [comment]
Agreed. I missed clearing the log action in the issue. Thanks for reminding that.
